### PR TITLE
feat(cache): Implement TTL cache for config

### PR DESCRIPTION
**Pull request by Jeevan-Kishore**
Closes: https://github.com/quintype/madrid/issues/5124

### CODE REVIEW

File: cache-constant.js [hunk 1]
```diff
@@ -0,0 +1,9 @@
+const CACHE_TIME = 240000;
+const MAX_CACHE = 200;
+const ENABLE_TTL_CACHE = false;
```
Comment (sharangj) [inline cache-constant.js:3]: Can you use environment variables?

File: index.js [hunk 2]
```diff
@@ -942,19 +950,36 @@ class Client {
...
+    if (ENABLE_TTL_CACHE) return this.getCacheConfig();
+
     if (this.config) return Promise.resolve(this.config);
 
     this.initialUpdateConfig = this.initialUpdateConfig || this.updateConfig();
 
     return this.initialUpdateConfig;
   }
 
+  async getCacheConfig() {
```
Comment (sharangj) [inline index.js:969]: Can you write documentation for this function for js docs?

**Review by sharangj — COMMENTED** [review]
Is it possible to share a sample ab or load test result on the PR?

File: cache-constant.js (hunk 1 above)
Comment (Jeevan-Kishore) [reply cache-constant.js:3]: Having it in a file is better scalable

**Comment by Jeevan-Kishore** [comment]
> Is it possible to share a sample ab or load test result on the PR?

## Load test

Users : 50
Urls : 8
Duration : 2 mins

## Before
image

## After
image

**Pull request merged**